File: app/src/main/java/com/akvelon/myfirstapp/lesson3/network/OkHttpProvider.java
package com.akvelon.myfirstapp.lesson3.network;

import android.support.annotation.NonNull;

import com.akvelon.myfirstapp.BuildConfig;
import com.facebook.stetho.okhttp3.StethoInterceptor;

import okhttp3.OkHttpClient;
import okhttp3.logging.HttpLoggingInterceptor;

public final class OkHttpProvider {

    private OkHttpProvider() {
    }

    @NonNull
    public static OkHttpClient provideClient() {
        final OkHttpClient.Builder builder = new OkHttpClient.Builder();
        builder.addInterceptor(new WeatherApiKeyInterceptor());
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(new HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY));
            builder.addNetworkInterceptor(new StethoInterceptor());
        }

        return builder.build();
    }

}

File: app/src/main/java/com/akvelon/myfirstapp/lesson9/data/network/OkHttpProvider.kt
package com.akvelon.myfirstapp.lesson9.data.network

import com.akvelon.myfirstapp.BuildConfig
import com.facebook.stetho.okhttp3.StethoInterceptor

import okhttp3.OkHttpClient
import okhttp3.logging.HttpLoggingInterceptor

object OkHttpProvider {

    fun provideClient(): OkHttpClient {
        val builder = OkHttpClient.Builder()
        builder.addInterceptor(WeatherApiKeyInterceptor())
        if (BuildConfig.DEBUG) {
            builder.addInterceptor(HttpLoggingInterceptor().setLevel(HttpLoggingInterceptor.Level.BODY))
            builder.addNetworkInterceptor(StethoInterceptor())
        }

        return builder.build()
    }

}